spec/writers/sparql_statement_writer_spec.rb

```diff
@@ -0,0 +1,212 @@
+# frozen_string_literal: true
+
+require 'rialto/etl/writers/sparql_statement_writer'
+require 'rialto/etl/readers/sparql_statement_reader'
+require 'rdf'
+require 'sparql'
+
+RSpec::Matchers.define :have_same_triples do |graph1|
+  match do |graph2|
+    # Compares statements only, ignoring graph names
+    return false unless graph1.size == graph2.size
+
+    graph1.each_triple { |s, o, p| return false unless graph2.has_triple?([s, o, p]) }
+    true
+  end
+
+  failure_message do |_|
+    'graphs do not contain same triples'
+  end
+
+  failure_message_when_negated do |_|
+    'graphs contain the same triples'
+  end
+
+  description do
+    'compares triples, ignoring graph names'
+  end
+end
+
+RSpec.describe Rialto::Etl::Writers::SparqlStatementWriter do
+  subject(:writer) { described_class.new({}) }
+
+  let(:repository) do
+    # Note: Create a named graph within repo by performing an insert. Delete cannot be
```

Comment:
It's not clear to me whether this is a TODO or an explanation of what is happening here. Can you add more explanation?